src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -327,27 +342,55 @@ private static <T extends SyntaxElement> boolean checkExperimentalSyntax(T eleme
 		return experimentalSyntax.isSatisfiedBy(experiments);
 	}
 
-	private static @NotNull DefaultExpression<?> getDefaultExpression(ExprInfo exprInfo, String pattern) {
-		DefaultExpression<?> expr;
-		// check custom default values first.
+	/**
+	 * Returns all {@link DefaultExpression}s from all the {@link ClassInfo}s embedded in {@code exprInfo} that pass all checks.
+	 *
+	 * @param exprInfo The {@link ExprInfo} to check for {@link DefaultExpression}s.
+	 * @param pattern The pattern used to create {@link ExprInfo}.
+	 * @return All available {@link DefaultExpression}s.
+	 * @throws SkriptAPIException If no {@link DefaultExpression}s pass, produces an error message for the reasoning of failure.
+	 */
+	static @NotNull List<DefaultExpression<?>> getDefaultExpressions(ExprInfo exprInfo, String pattern) {
```

Comment:
I don't think it's ideal that we grab/check all the default expressions even if we don't need them.
Do you think it would be possible to keep an adapted singular `getDefaultExpression` method? Essentially, if ExprInfo only has a single class, we can call that and avoid instantiating the map and list. Otherwise, the `getDefaultExpressions` method (which can just call the singular method). The singular method can return some sort of retrieval result (that includes either the expression or an error type)